### PR TITLE
Refactor FakeQuantizeLayerTest, ExtractImagePatchesTest, GatherNDLayerTest, GatherTreeLayerTest

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Refactor FakeQuantizeLayerTest, ExtractImagePatchesTest, GatherNDLayerTest, GatherTreeLayerTest
 - Update cpu plugin instantiation of these tests to use new implementation

### Tickets:
 - [CVS-111364](https://jira.devtools.intel.com/browse/CVS-111364)

### CODE REVIEW

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+namespace test {
+typedef std::tuple<
+        size_t,                     // fake quantize levels
+        std::vector<size_t>,        // fake quantize inputs shape
+        std::vector<float>,         // fake quantize (inputLow, inputHigh, outputLow, outputHigh) or empty for random
+        ov::op::AutoBroadcastSpec   // fake quantize broadcast mode
+> fqSpecificParams;
+typedef std::tuple<
+        fqSpecificParams,
+        ov::element::Type,                                         // Net precision
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp:35]: // Net precision -> model_type

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_nd.hpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+
+typedef std::tuple<
+        std::vector<InputShape>,  // Input shapes
+        ov::Shape,                // Indices shape
+        int,                      // batch dims
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_nd.hpp:19]: // batch dims -> Batch dim (because there is only one value)

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_tree.hpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+#include "common_test_utils/test_enums.hpp"
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+using GatherTreeParamsTuple = typename std::tuple<
+        ov::Shape,                         // Input tensors shape
+        ov::test::utils::InputLayerType,   // Secondary input type
+        ov::element::Type,                 // Network precision
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_tree.hpp:19]: Model type

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 4]
```diff
@@ -455,7 +455,9 @@ ov::runtime::Tensor generate(const std::shared_ptr<ngraph::op::v1::GatherTree>&
             return ov::test::utils::create_and_fill_tensor(elemType, targetShape, inGenData.range, inGenData.start_from, inGenData.resolution, inGenData.seed);
         }
         default:
-            return generate(std::dynamic_pointer_cast<ov::Node>(node), port, elemType, targetShape);
+            InputGenerateData inGenData;
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:458]: Why the change was required?

File: src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    ov::element::Type model_type, indices_type;
+    int batch_dims;
+    std::tie(shapes, indices_shape, batch_dims, model_type, indices_type, targetDevice) = this->GetParam();
+    init_input_shapes(shapes);
+
+    auto param = std::make_shared<ov::op::v0::Parameter>(model_type, inputDynamicShapes.front());
+
+    auto gather = ngraph::builder::makeGatherND(param, indices_shape, indices_type, batch_dims);
+
+    auto result = std::make_shared<ngraph::opset1::Result>(gather);
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp:52]: ngraph to be removed

File: src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    function = std::make_shared<ngraph::Function>(result, ov::ParameterVector{param}, "gatherND");
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp:53]: same

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:458]: test requires changes in input data generation

Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:458]: generate(...) creates tensor with default values. I need `range` equal to `maxBeamIndx`

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp (hunk 1 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp]: done

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_nd.hpp (hunk 2 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_nd.hpp]: done

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_tree.hpp (hunk 3 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/gather_tree.hpp]: done

File: src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp (hunk 5 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp]: done

File: src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp (hunk 6 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/single_op/gather_nd.cpp]: done

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+// seed selected using current cloc time
+#define USE_CLOCK_TIME 1
+// seed started from default value, and incremented every time using big number like 9999
+#define USE_INCREMENTAL_SEED 2
+
+/**
+ * redefine this seed to reproduce issue with given seed that can be read from gtest logs
+ */
+#define BASE_SEED   123
+#define OV_SEED 123
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp:23]: Hm. Looks like hardcode. Let's ask fake quantize @eshoguli . @eshoguli What do you think about this tests? It it correct or should be removed?

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 4 above)
Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:458]: Could you please explain the details? Anyway, we can change the `generate` signature as was discussed

File: src/tests/functional/shared_test_classes/src/single_op/fake_quantize.cpp [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+            {fq_direct_arg[3]});
+    }
+
+    auto result = std::make_shared<ov::op::v0::Result>(fq);
+    function = std::make_shared<ov::Model>(result, ov::ParameterVector{param}, "fakeQuantize");
+}
+
+void FakeQuantizeLayerTest::update_seed() {
+    if (BASE_SEED == USE_CLOCK_TIME) {
+        seed = std::chrono::system_clock::now().time_since_epoch().count();
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/single_op/fake_quantize.cpp:98]: @eshoguli just label to you

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp (hunk 7 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/fake_quantize.hpp]: I removed all hardcode because it doesn't work indeed

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:458]: I aligned behavior with legacy tests. Otherwise tests fail

**Review by iefode — APPROVED** [review]
